algorithms/unit_tests/TestStdAlgorithmsTeamFindFirstOf.cpp

```diff
@@ -0,0 +1,290 @@
+    auto myRowViewFrom = Kokkos::subview(m_dataView, myRowIndex, Kokkos::ALL());
+    auto myRowSearchedSeqView =
+        Kokkos::subview(m_searchedSequencesView, myRowIndex, Kokkos::ALL());
+
+    switch (m_apiPick) {
+      case 0: {
+        auto it = KE::find_first_of(
+            member, KE::cbegin(myRowViewFrom), KE::cend(myRowViewFrom),
+            KE::cbegin(myRowSearchedSeqView), KE::cend(myRowSearchedSeqView));
+        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::cbegin(myRowViewFrom), it);
+        });
+
+        break;
+      }
+
+      case 1: {
+        auto it =
+            KE::find_first_of(member, myRowViewFrom, myRowSearchedSeqView);
+        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::begin(myRowViewFrom), it);
+        });
+
+        break;
+      }
+
+      case 2: {
+        auto it = KE::find_first_of(member, KE::cbegin(myRowViewFrom),
+                                    KE::cend(myRowViewFrom),
+                                    KE::cbegin(myRowSearchedSeqView),
+                                    KE::cend(myRowSearchedSeqView), m_binaryOp);
+        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::cbegin(myRowViewFrom), it);
+        });
+
+        break;
+      }
+
+      case 3: {
+        auto it = KE::find_first_of(member, myRowViewFrom, myRowSearchedSeqView,
+                                    m_binaryOp);
+        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::begin(myRowViewFrom), it);
+        });
+
+        break;
+      }
+    }
+  }
+};
+
+template <class LayoutTag, class ValueType>
+void test_A(const bool sequencesExist, std::size_t numTeams,
+            std::size_t numCols, int apiId) {
+  /* description:
+     use a rank-2 view randomly filled with values,
+     and run a team-level find_first_of
+   */
+
+  // -----------------------------------------------
+  // prepare data
+  // -----------------------------------------------
+  // create a view in the memory space associated with default exespace
+  // with as many rows as the number of teams and fill it with random
+  // values from an arbitrary range.
+  const ValueType lowerBound{5}, upperBound{523};
+  auto [dataView, dataViewBeforeOp_h] = create_random_view_and_host_clone(
+      LayoutTag{}, numTeams, numCols, Kokkos::pair{lowerBound, upperBound},
+      "dataView");
+
+  // create a view that stores a sequence to found a value from in dataView. If
+  // sequencesExist == true it is filled base on dataView content, to allow
+  // find_first_of to actually find anything. If sequencesExist == false it is
+  // filled with random values greater than upperBound
+  const std::size_t halfCols = (numCols > 1) ? ((numCols + 1) / 2) : (1);
```

Comment:
why (1) ?

Reply:
Are you asking why `(1)` instead of `1`, or why I set halfCols to 1 when numcols == 1?